framework/core/src/User/Command/RequestPasswordResetHandler.php

```diff
@@ -9,77 +9,33 @@
 
 namespace Flarum\User\Command;
 
-use Flarum\Http\UrlGenerator;
-use Flarum\Mail\Job\SendRawEmailJob;
-use Flarum\Settings\SettingsRepositoryInterface;
-use Flarum\User\PasswordToken;
-use Flarum\User\UserRepository;
+use Flarum\User\Job\RequestPasswordResetJob;
 use Illuminate\Contracts\Queue\Queue;
 use Illuminate\Contracts\Validation\Factory;
-use Illuminate\Database\Eloquent\ModelNotFoundException;
 use Illuminate\Validation\ValidationException;
-use Symfony\Contracts\Translation\TranslatorInterface;
 
 class RequestPasswordResetHandler
 {
-    /**
-     * @var UserRepository
-     */
-    protected $users;
-
-    /**
-     * @var SettingsRepositoryInterface
-     */
-    protected $settings;
-
     /**
      * @var Queue
      */
     protected $queue;
 
-    /**
-     * @var UrlGenerator
-     */
-    protected $url;
-
-    /**
-     * @var TranslatorInterface
-     */
-    protected $translator;
-
     /**
      * @var Factory
      */
     protected $validatorFactory;
 
-    /**
-     * @param UserRepository $users
-     * @param SettingsRepositoryInterface $settings
-     * @param Queue $queue
-     * @param UrlGenerator $url
-     * @param TranslatorInterface $translator
-     * @param Factory $validatorFactory
-     */
-    public function __construct(
-        UserRepository $users,
-        SettingsRepositoryInterface $settings,
-        Queue $queue,
-        UrlGenerator $url,
-        TranslatorInterface $translator,
-        Factory $validatorFactory
-    ) {
-        $this->users = $users;
-        $this->settings = $settings;
+    public function __construct(Queue $queue, Factory $validatorFactory)
+    {
         $this->queue = $queue;
-        $this->url = $url;
-        $this->translator = $translator;
         $this->validatorFactory = $validatorFactory;
     }
 
     /**
      * @param RequestPasswordReset $command
-     * @return \Flarum\User\User
-     * @throws ModelNotFoundException
+     * @return void
+     * @throws ValidationException
      */
     public function handle(RequestPasswordReset $command)
```

Comment:
Yeah, let's kill this and go straight to controller. It's not an API route, so we don't really lose abstractability, and the nested controller > handler > job is a bit too much complexity IMO.